app/controllers/application_controller.rb

```diff
@@ -16,6 +17,18 @@ def after_sign_in_path_for(user)
 
   private
 
+  def set_locale
+    I18n.locale = if params[:locale].to_s.to_sym.in?(I18n.available_locales)
+      params[:locale]
+    else
+      I18n.default_locale
+    end
+  end
+
+  def default_url_options(options = {})
+    options.merge({locale: I18n.locale})
+  end
```

Comment:
This method is making me a little nervous. 

1. `developer_url(Developer.first)` is now adding the locale `en` to the URL. If the current locale is English then we shouldn't add a locale at all. For example, clicking from the homepage to a developer adds in `/en/*`.
2. This looks to be the cause of the broken tests and adding needing to set the `id:` param for some path generation.

Reply:
In order to avoid set locale to all the `*_url`, `*_path` in whole application, I take this from [Rails guide](https://guides.rubyonrails.org/i18n.html#setting-the-locale-from-url-params).
I have made some changes to avoid adding the locale en to the URL.
Please review the newer version.👍

Reply:
This also not needed after set locale nil in routes.rb.